File: problem_set/palindrome-permutation-lcci/palindrome-permutation-lcci.java

```
// @Title: 回文排列 (Palindrome Permutation LCCI)
// @Author: robert.sunq
// @Date: 2021-08-02 20:59:10
// @Runtime: 0 ms
// @Memory: 36.2 MB

class Solution {
    public boolean canPermutePalindrome(String s) {
        int[] dict = new int[256];
        for(Character c : s.toCharArray()){
            dict[c]++;
        }
        int res = 0;
        for(int i = 0;i<256;i++){
            if(dict[i]%2 == 1) res++;
        }
        return res<2;
    }
}

File: problem_set/palindrome-permutation-lcci/palindrome-permutation-lcci.cpp

```
// @Title: 回文排列 (Palindrome Permutation LCCI)
// @Author: robert.sunq
// @Date: 2020-03-23 17:02:06
// @Runtime: 0 ms
// @Memory: 7.7 MB

class Solution {
public:
    bool canPermutePalindrome(string s) {
        int flag = 0;
        if(s.size() == 0) return true;
        int an[256] = {0};
        for(int i=0;i<s.size();i++){
            an[s[i]]++;
        }

        for(int i=0;i<256;i++){
            if(an[i]%2) flag++;
            if(flag >1) return false;
        }
        return true;
    }
};
